mvvm_cards/src/shared/Modal.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from "react";
+import ReactDOM from "react-dom";
+
+type Props = {
+  onDecline: () => void;
+  onConfirm: () => void;
+  children: React.ReactNode[] | React.ReactNode
+}
+
+export default function Modal(props: Props) {
+  const { children, onDecline, onConfirm } = props;
+
+  const bodyEl = document.querySelector("body");
+
+  return (
+    ReactDOM.createPortal(
```

Comment:
in general, it should be like you described but I thought that in this case, it will be redundant because we need to pass the _open_ prop through few parent components. So that's why I made it so simple.

if you want me to do it so I can also move this 'showDialog' property to the AppModel and in this case we won't need to pass this property through all parent components.

But, to be honest, I'd like rather create a separated CardsStore (eg. CardsModel) where we could keep al info regarding our particular CardModel

I don't like the solution to push everything to the one AppModel. I described the reason before

Reply:
Since the code is keeping the dialog around, I would like the visibility of the dialog controlled by wether or not it is holding on to an ICard object.  So ```dialog.card = null``` would automatially hide the dialog and setting to to a card would automatically show it.

Reply:
It could be done for sure, but in this case, this Dialog won't be re-usable because of the strict Card content dependency. 
The common use case for re-usable Dialog is just added the Show  boolean property and show the dialog regarding this prop value.

But if you want to show/hide this dialog based on the content (if the content presented) so I can do it as well. I'd like to do it generically, not for cards only
We can do it via _children_ property. If it exists - we'll show the dialog, if not - hide. In this case, we can use Dialog not only for Card but for any other content type as well
Anyway, it will add additional complexity

Reply:
Done, but I'm not sure that it's exactly what you wanted, so, could you please check it and if it isn't that you expected, I'll change it in my free time.
Btw. I don't much like these changes that I did because it's a bit redundant (just my opinion)

Maybe you wanted to add the state Show/Hide directly to the ModalDialog and based on the children property (or Card) it should change its show property to false/true?